src/gepa/examples/anymaths-bench/train_anymaths.py

```diff
@@ -0,0 +1,124 @@
+    valset = train_split[len(train_split) // 2 :]
+    testset = test_split * 5
+
+    return trainset, valset, testset
+
+
+if __name__ == "__main__":
+    import argparse
+    from pathlib import Path
+
+    import litellm
+
+    from gepa import optimize
+    from gepa.adapters.anymaths_adapter import AnyMathsAdapter
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--model_name", type=str, default="ollama/qwen3:4b")  # only Ollama models are supported
+    parser.add_argument("--api_base", type=str, default="http://localhost:11434")
+    parser.add_argument("--max_litellm_workers", type=int, default=10)
+    parser.add_argument("--anymaths_dset_name", type=str, default="openai/gsm8k")
+    parser.add_argument("--budget", type=int, default=5, help="The budget for the optimization process.")
+    parser.add_argument(
+        "--reflection_lm", type=str, default="ollama/qwen3:8b", help="The name of the reflection LM to use."
+    )
+    parser.add_argument(
+        "--reflection_minibatch_size", type=int, default=3, help="The size of the minibatch for the reflection LM."
+    )
+    parser.add_argument(
+        "--seed", type=int, default=0, help="The seed for the random number generator for reproducibility."
+    )
+    args = parser.parse_args()
+
+    INSTRUCTION_PROMPT_PATH = Path(__file__).parent / "prompt-templates/instruction_prompt.txt"
+
+    seed_instruction = INSTRUCTION_PROMPT_PATH.read_text()
+
+    trainset, valset, testset = init_dataset(args.anymaths_dset_name)
+
+    trainset = trainset[:50]  # Limit to 50 samples for demo purposes
+    valset = valset[:50]  # Limit to 50 samples for demo purposes
+    testset = testset[:50]  # Limit to 50 samples for demo purposes
+
+    print(f"Train set size: {len(trainset)}")
+    print(f"Validation set size: {len(valset)}")
+    print(f"Test set size: {len(testset)}")
+
+    reflection_lm_name = args.reflection_lm
+
+    def reflection_lm(prompt: str):
+        """Call the reflection language model with the given prompt and return its content string."""
+        response = litellm.completion(
+            model=reflection_lm_name,
+            messages=[{"role": "user", "content": prompt}],
+        )
+        return response.choices[0].message.content
+
+    adapter_model = args.model_name
+    api_base = args.api_base
+    max_litellm_workers = args.max_litellm_workers
+    budget = args.budget
+    reflection_minibatch_size = args.reflection_minibatch_size
+    seed = args.seed
+
+    optimized_results = optimize(
+        seed_candidate={"instruction_prompt": seed_instruction},
+        trainset=trainset,
+        valset=valset,
+        adapter=AnyMathsAdapter(model=adapter_model, api_base=api_base, max_litellm_workers=max_litellm_workers),
+        reflection_lm=reflection_lm,
+        reflection_minibatch_size=reflection_minibatch_size,
+        perfect_score=1,
+        skip_perfect_score=False,
+        use_wandb=False,
+        num_iters=budget,
```

Comment:
I just merged a PR removing the num_iters argument. Can you convert this into num rollouts instead?